src/main/kotlin/com/dansplugins/factionsystem/dynmap/MfDynmapService.kt

```diff
@@ -51,265 +76,170 @@ class MfDynmapService(private val plugin: MedievalFactions) {
-                                            faction.id,
-                                            {
-                                                val areaMarker = realmMarkerSet.createAreaMarker(
-                                                    "realm_${faction.id.value}_${worldId}_$index",
-                                                    faction.name,
-                                                    false,
-                                                    world.name,
-                                                    corners.map { (x, _) -> x * 16.0 }.toDoubleArray(),
-                                                    corners.map { (_, z) -> z * 16.0 }.toDoubleArray(),
-                                                    false
-                                                )
-                                                areaMarker.label = faction.name
-                                                val color = Integer.decode(faction.flags[plugin.flags.color])
-                                                areaMarker.setFillStyle(0.0, color)
-                                                areaMarker.setLineStyle(4, 1.0, color)
-                                                areaMarker.description = factionInfo
-                                                val factionMarkers = factionMarkersByFactionId[faction.id] ?: listOf()
-                                                factionMarkersByFactionId[faction.id] = factionMarkers + areaMarker
-                                            }
-                                        ) { runTask(plugin) }
-                                    }
-                                }
-                            ) { runTaskAsynchronously(plugin) }
-                        }
-                    }
-                ) { runTask(plugin) }
-            }
-        }) { runTaskAsynchronously(plugin) }
+        createUpdateTask(faction.id, { updateFactionClaims(faction, claimsMarkerSet, claimService) }) { runTaskAsynchronously(plugin) }
+        createUpdateTask(faction.id, { updateFactionRealm(faction, realmMarkerSet, claimService) }) { runTaskAsynchronously(plugin) }
     }
 
-    private fun getCorners(points: List<Point>): List<Point> {
-        val corners = mutableListOf<Pair<Int, Int>>()
-        for (i in points.indices) {
-            val (prevX, prevZ) = if (i > 0) points[i - 1] else points.last()
-            val (nextX, nextZ) = if (i < points.lastIndex) points[i + 1] else points.first()
-            if (prevX != nextX && prevZ != nextZ) {
-                corners.add(points[i])
-            }
+    private fun updateFactionClaims(faction: MfFaction, claimsMarkerSet: MarkerSet, claimService: MfClaimService) {
+        val claims = claimService.getClaims(faction.id)
+        if (debug) {
+            plugin.logger.info("Fetched ${claims.size} claims for faction ${faction.name}")
+        }
+        val factionInfo = FactionInfoBuilder(plugin).build(faction)
```

Comment:
Rather than instantiate a new instance of FactionInfoBuilder every time this method is called we should use a class instance

Reply:
Addressed